### PR TITLE
feat(inputs.mysql): Add support for replica status

### LINKED ISSUE #15748: feat(inputs.mysql): Add support gather replica status
### Info
Due to from MYSQL 8.4 removed `show slave status` command and replace by `show replica status` command. this cause issue when collect from MYSQL 8.4 release and further.
More: https://dev.mysql.com/doc/relnotes/mysql/8.4/en/news-8-4-0.html#mysqld-8-4-0-deprecation-removal
### Relevant telegraf.conf

```toml
[[inputs.mysql]]
  metric_version = 2
  gather_all_slave_channels = true
  servers = ['root:test@tcp(127.0.0.1:34306)/']
  gather_slave_status = true
```

### Logs from Telegraf

```text
2024-08-16T01:11:30Z E! [inputs.mysql] Error in plugin: Error 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'SLAVE STATUS' at line 1
```

### System info

Telegraf 1.31.3 (git: HEAD@ecf94b12)

### Docker

_No response_

### Steps to reproduce

1. Setup MYSQL 8.4 replication
2. Enable collect slave status with option `gather_slave_status`
3. Check log and see issue 

### Expected behavior

- Have metric related replica in mysql 8.4 like 
```
Replica_IO_Running
Replica_SQL_Running
```

### Actual behavior

2024-08-16T01:11:30Z E! [inputs.mysql] Error in plugin: Error 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'SLAVE STATUS' at line 1

### Additional info

_No response_
```[tasklist]
### Tasks
- [ ] https://github.com/influxdata/telegraf/pull/15749
```

### PR DESCRIPTION
**Pull request by hungnt1**
## Summary
Add replica status command when collect mysql

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #15748

### CODE REVIEW

File: plugins/inputs/mysql/mysql.go [hunk 1]
```diff
@@ -77,12 +78,18 @@ func (*Mysql) SampleConfig() string {
...
+		case m.MariadbDialect && m.GatherReplicaStatus:
+			return replicaStatusQueryMariadb
+		case m.MariadbDialect:
+			return slaveStatusQueryMariadb
+		case m.GatherReplicaStatus:
+			return replicaStatusQuery
+		default:
+			return slaveStatusQuery
+		}
+	}()
```
Comment (srebhan) [inline plugins/inputs/mysql/mysql.go:92]: Why is this needed instead of directly using the switch statement? Are you sure you didn't use some weird AI tool to create this?

File: plugins/inputs/mysql/README.md [hunk 2]
```diff
@@ -260,7 +264,8 @@ the single-source replication is on. If the multi-source replication is set,
 then everything works differently, this metric does not work with multi-source
 replication, unless you set `gather_all_slave_channels = true`. For MariaDB,
 `mariadb_dialect = true` should be set to address the field names and commands
-differences.
+differences. If enable `gather_replica_status` metrics gather from command
```
Comment (srebhan) [inline plugins/inputs/mysql/README.md:267]: Please remove the trailing whitespace to make the linter happy!
```suggestion
differences. If enable `gather_replica_status` metrics gather from command
```

**Review by srebhan — COMMENTED** [review]
Thanks for your contribution @hungnt1! Two comments from my side...

File: plugins/inputs/mysql/mysql.go (hunk 1 above)
Comment (hungnt1) [reply plugins/inputs/mysql/mysql.go:92]: Just ensure string output. We can use directly switch.

Comment (hungnt1) [reply plugins/inputs/mysql/mysql.go]: Updated

**Review by srebhan — APPROVED** [review]
Awesome! Thanks @hungnt1 for this nice feature!

**Review by powersj — APPROVED** [review]
